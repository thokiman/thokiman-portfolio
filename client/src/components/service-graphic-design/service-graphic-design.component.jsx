import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { CSSTransitionGroup } from "react-transition-group";

import {
  RightServiceContainerGd,
  RightServiceContentGd,
} from "./service-graphic-design.styles";
import ServiceGeneralContent from "../service-general-content/service-general-content.component";
import ServiceIconContent from "../service-icon-content/service-icon-content.component";
import { selectArtService } from "../../redux/service/service.selectors";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";

export const ServiceGraphicDesign = ({
  isSideBarHidden,
  art: { title, description, items },
}) => {
  return (
    <RightServiceContainerGd $issidebarhidden={isSideBarHidden}>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={800}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <RightServiceContentGd>
          <ServiceGeneralContent title={title} description={description} />
          <ServiceIconContent items={items} />
        </RightServiceContentGd>
      </CSSTransitionGroup>
    </RightServiceContainerGd>
  );
};
const mapStateToProps = createStructuredSelector({
  art: selectArtService,
  isSideBarHidden: selectIsSideBarHidden,
});
export default connect(mapStateToProps)(ServiceGraphicDesign);
