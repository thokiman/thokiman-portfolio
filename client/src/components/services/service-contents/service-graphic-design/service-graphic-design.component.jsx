import useWindowDimensions from 'components/hooks/window-dimensions/useWindowDimensions.component';
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectIsSideBarActive } from '../../../../redux/header/header.selectors';
import { selectArtService } from '../../../../redux/service/service.selectors';
import ServiceGeneralContent from '../../service-commons/service-general-content/service-general-content.component';
import ServiceIconContent from '../../service-commons/service-icon-content/service-icon-content.component';
import {
  RightServiceContainerGd,
  RightServiceContentGd,
} from './service-graphic-design.styles';
import { measureRightPageContainerGdProps } from './service-graphic-design.utils.style';

export const ServiceGraphicDesign = ({
  isSideBarActive,
  art: { title, description, items },
}) => {
  const { viewWidth, viewHeight } = useWindowDimensions();
  return (
    <RightServiceContainerGd
      $measurerightpagecontainergd={measureRightPageContainerGdProps(
        viewWidth,
        viewHeight
      )}
      className='right-service-container'
      $issidebaractive={isSideBarActive}
    >
      <RightServiceContentGd>
        <ServiceGeneralContent title={title} description={description} />
        <ServiceIconContent items={items} />
      </RightServiceContentGd>
    </RightServiceContainerGd>
  );
};
const mapStateToProps = createStructuredSelector({
  art: selectArtService,
  isSideBarActive: selectIsSideBarActive,
});
export default connect(mapStateToProps)(ServiceGraphicDesign);
