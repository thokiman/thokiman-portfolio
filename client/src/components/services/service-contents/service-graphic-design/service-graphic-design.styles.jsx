import styled, { css } from 'styled-components';

const backgroundColorRightServiceContainerGd = 'rgba(18, 129, 148, 0.75)';
const maxWidthSmartphone = 800;
const ServicePhotographerIsNotActivated = css`
  display: none;
`;
const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return ServicePhotographerIsNotActivated;
  return null;
};

export const RightServiceContainerGd = styled.div`
  background-color: ${backgroundColorRightServiceContainerGd};
  width: 66.3vw;
  height: 68vh;
  padding: 10px 20px;
  position: relative;
  right: 4vw;
  display: flex;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    overflow-y: hidden;
    width: 100%;
    right: 0;
    height: 175vh;
    ${getSideBarActive}

    @media only screen and (max-width: 550px) {
      height: 160vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 823px) {
      height: 110vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 812px) {
      height: 110vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 736px) {
      height: 125vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 470px) and (max-height: 667px) {
      height: 135vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 350px) and (max-height: 568px) {
      height: 145vh;
      ${getSideBarActive}
    }
    @media only screen and (max-width: 300px) {
      height: 135vh;
      ${getSideBarActive}
    }
  }
`;

export const RightServiceContentGd = styled.div`
  width: 95%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  margin: auto 0;
  padding: 20px;
`;

RightServiceContainerGd.displayName = 'RightServiceContainerGd';
RightServiceContentGd.displayName = 'RightServiceContentGd';
