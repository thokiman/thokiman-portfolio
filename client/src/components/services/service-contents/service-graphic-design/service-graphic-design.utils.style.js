import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
} from '../../../../utils/common.component.utils';

export const measureRightPageContainerGdProps = (viewWidth, viewHeight) => {
  const rightServiceContainerGdHeightOffset = 0;
  const rightServiceContainerGdTopOffset = 0;

  let rightServiceContainerProps;

  if (hasScreenSize1(viewWidth)) {
    console.log('service-graphic-design', 1);
    rightServiceContainerProps = {
      height: 165 + rightServiceContainerGdHeightOffset,
      top: -100 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('service-graphic-design', 2);
    rightServiceContainerProps = {
      height: 165 + rightServiceContainerGdHeightOffset,
      top: -65 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('service-graphic-design', 3);
    rightServiceContainerProps = {
      height: 110 + rightServiceContainerGdHeightOffset,
      top: -70 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('service-graphic-design', 4);
    rightServiceContainerProps = {
      height: 105 + rightServiceContainerGdHeightOffset,
      top: -75 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('service-graphic-design', 5);
    rightServiceContainerProps = {
      height: 125 + rightServiceContainerGdHeightOffset,
      top: -75 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('service-graphic-design', 6);
    rightServiceContainerProps = {
      height: 130 + rightServiceContainerGdHeightOffset,
      top: -85 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('service-graphic-design', 7);
    rightServiceContainerProps = {
      height: 145 + rightServiceContainerGdHeightOffset,
      top: 0 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('service-graphic-design', 8);
    rightServiceContainerProps = {
      height: 145 + rightServiceContainerGdHeightOffset,
      top: -70 + rightServiceContainerGdTopOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('service-graphic-design', 9);
    rightServiceContainerProps = {
      height: 135 + rightServiceContainerGdHeightOffset,
      top: -67 + rightServiceContainerGdTopOffset,
    };
  } else {
    console.log('service-graphic-design', 10);
    rightServiceContainerProps = {
      height: 175,
      top: -110,
    };
  }

  return rightServiceContainerProps;
};
