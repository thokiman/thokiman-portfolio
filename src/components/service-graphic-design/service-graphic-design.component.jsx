import React from "react";

import ServiceGeneralContent from "../service-general-content/service-general-content.component";
import ServiceIconContent from "../service-icon-content/service-icon-content.component";
import "./service-graphic-design.styles.scss";

const ServiceGraphicDesign = ({ art: { title, description, items } }) => {
  return (
    <div className="right-service-container-gd">
      <div className="right-service-content-gd">
        <ServiceGeneralContent title={title} description={description} />
        <ServiceIconContent items={items} />
      </div>
    </div>
  );
};

export default ServiceGraphicDesign;
