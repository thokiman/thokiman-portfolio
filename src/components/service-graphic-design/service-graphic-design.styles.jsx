import styled from "styled-components";

export const RightServiceContainerGd = styled.div`
  background-color: rgba(18, 129, 148, 0.75);
  width: 66.3vw;
  height: 68vh;
  padding: 10px 20px;
  position: relative;
  right: 4vw;
  display: flex;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;
`;

export const RightServiceContentGd = styled.div`
  width: 95%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  margin: auto 0;
  padding: 20px;
`;

RightServiceContainerGd.displayName = "RightServiceContainerGd";
RightServiceContentGd.displayName = "RightServiceContentGd";
