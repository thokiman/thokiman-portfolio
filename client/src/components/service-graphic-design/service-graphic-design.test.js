import React from "react";
import { shallow } from "enzyme";

import { ServiceGraphicDesign } from "./service-graphic-design.component";

describe("ServiceGraphicDesign component", () => {
  let wrapper;

  const mockFalseBool = false;
  const mockText = "";
  const mockArray = [];
  beforeEach(() => {
    const mockProps = {
      isSideBarHidden: mockFalseBool,
      art: { title: mockText, description: mockText, items: mockArray },
    };
    wrapper = shallow(<ServiceGraphicDesign {...mockProps} />);
  });

  it("should render ServiceGraphicDesign component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
