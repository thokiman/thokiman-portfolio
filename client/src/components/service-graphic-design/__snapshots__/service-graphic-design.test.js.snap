// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceGraphicDesign component should render ServiceGraphicDesign component 1`] = `ShallowWrapper {}`;
